Sources/App/Scenes/WindowScenesManager.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import Shared
+import UIKit
+
+final class WindowScenesManager {
+    static var shared = WindowScenesManager()
+    private var windowSizeObservers: [WindowSizeObserver] = []
+
+    func sceneDidBecomeActive(_ scene: UIWindowScene) {
+        if #available(macCatalyst 16.0, *) {
+            configureSceneSize(scene)
+            startObservingScene(scene)
+        }
+    }
+
+    private func startObservingScene(_ scene: UIWindowScene) {
+        let observer = WindowSizeObserver(windowScene: scene)
+        windowSizeObservers.append(observer)
+    }
+
+    func didDiscardScene(_ scene: UIScene) {
+        windowSizeObservers.removeAll(where: { $0.observedScene == scene })
+    }
+
+    private func sceneFrameIsValid(_ sceneFrame: CGRect, screenSize: CGSize) -> Bool {
+        sceneFrame.height <= screenSize.height && sceneFrame.width <= screenSize.width
+    }
+
+    private func adjustedSystemFrame(
+        _ systemFrame: CGRect,
+        for screenSize: CGSize,
+        numberOfConnectedScenes: Int
+    ) -> CGRect {
+        guard numberOfConnectedScenes > 1 else { return systemFrame }
```

Comment:
what does this do? can you add a comment?

Reply:
i'm still unsure what the connected scene count is doing here. why not restore the frame if there's 1 connected scene?

Reply:
So basically, if you have just one scene, ir will recover the latest window size, if you have more than one scene and both were positioned in the same coordinates, it will make them a little bit offset (cascade effect) so one is not hidden.
I got this idea from one article and I thought it would be a nice touch, what do you think?

Reply:
ahhh i understand now. yeah that is a good effect!